### PR TITLE
Remove { collection } route binding

### PR DESCRIPTION
**Pull request by ryanmitchell**
I'm having issues with the `{ collection }` route binding when using Statamic alongside another package. 

So as suggested [here](https://github.com/statamic/cms/issues/4108) I've removed the collection binding from all routes that use it.

I'm happy to do the leg work on any changes you want to this - so just direct me.

### CODE REVIEW

**Comment by jasonvarga** [comment]
I appreciate this PR and think it's a good change to have, but I think it might be a potential breaking change for addons that may be relying on these bindings in the control panel.

Could we instead just provide the bindings to control panel routes?

Like we do for the API:

https://github.com/statamic/cms/blob/38f1116e2ac6c842baad24692225edae89481b7a/src/Providers/RouteServiceProvider.php#L89-L91

Have one of these:

```php
if (! $this->isCpRoute($route)) {
  return $handle;
}
```

File: src/Http/Controllers/API/CollectionEntriesController.php [hunk 1]
```diff
@@ -16,6 +17,12 @@ public function index($collection)
     {
         $this->abortIfDisabled();
 
+        $collectionHandle = $collection;
+        $collection = Collection::findByHandle($collection);
+        if (! $collection) {
+            throw new NotFoundHttpException("Collection [$collectionHandle] not found.");
+        }
+
```
Comment (edalzell) [inline src/Http/Controllers/API/CollectionEntriesController.php:25]: `throw_unless($collection = Collection::findByHandle($collection))`?

File: src/Http/Controllers/API/CollectionEntriesController.php [hunk 2]
```diff
@@ -29,6 +36,12 @@ public function show($collection, $handle)
...
```
Comment (edalzell) [inline src/Http/Controllers/API/CollectionEntriesController.php:44]: same as above?

**Comment by ryanmitchell** [comment]
@jasonvarga well that would have been a lot less work 🤣 new PR incoming

**Comment by jasonvarga** [comment]
Sorry 😬

**Pull request closed without merging**